runtime/compiler/env/CHTable.hpp

```diff
@@ -127,10 +127,9 @@ class TR_PatchNOPedGuardSiteOnClassPreInitialize : public TR::PatchNOPedGuardSit
       TR_FrontEnd *fe, TR_PersistentMemory *, char *sig, uint32_t sigLen, uint8_t *loc, uint8_t *dest, OMR::RuntimeAssumption **sentinel);
    static uintptrj_t hashCode(char *sig, uint32_t sigLen);
 
-   virtual void reclaim() { jitPersistentFree((void*)_key); }
+   virtual void reclaim();
```

Comment:
I am wondering if it would be wises for you to add a line or two of doxygen to note that the assumption persistent resources should only be freed in reclaim since the assumption can still be looked at/processed until the reclaim is run during deallocation. Might help make sure someone else doesn't make the same mistake in a compensate handler in the future?

Reply:
Fixed in [force push](https://github.com/eclipse/openj9/compare/f73530051f1fd15719269f628d2fe20c9065869f..5b57a0b92d691020cd06040de2ce7f1d86a74306). Note the base API lives in OMR. I've opened eclipse/omr#4256 to add the documentation there.